go/vt/vtgate/evalengine/fn_time.go

```diff
@@ -330,7 +330,14 @@ type builtinConvertTz struct {
 
 var _ IR = (*builtinConvertTz)(nil)
 
-func convertTz(dt datetime.DateTime, from, to *time.Location) (datetime.DateTime, bool) {
+func convertTz(dt datetime.DateTime, from, to *time.Location, now time.Time) (datetime.DateTime, bool) {
+	t := dt.ToStdTime(now)
+	lowerBoundTz := time.Date(1970, 01, 01, 0, 0, 0, 0, time.UTC)
+	upperBoundTz := time.Date(3001, 1, 19, 3, 14, 7, 99999, time.UTC)
+	if t.Before(lowerBoundTz) || t.After(upperBoundTz) {
+		return dt, true
+	}
+
 	buf := datetime.DateTime_YYYY_MM_DD_hh_mm_ss.Format(dt, datetime.DefaultPrecision)
 	ts, err := time.ParseInLocation(time.DateTime, hack.String(buf), from)
```

Comment:
We already parsed it into a Go Time here. So the above conversion seems wasteful to me. I think what we can do is after we've done this parse, is check against the unix timestamp instead? So something like this instead which seems a lot simpler?

```patch
diff --git i/go/vt/vtgate/evalengine/fn_time.go w/go/vt/vtgate/evalengine/fn_time.go
index ecb1fedc13..ac1f03b589 100644
--- i/go/vt/vtgate/evalengine/fn_time.go
+++ w/go/vt/vtgate/evalengine/fn_time.go
@@ -341,6 +341,9 @@ func convertTz(dt datetime.DateTime, from, to *time.Location) (datetime.DateTime
        if err != nil {
                return datetime.DateTime{}, false
        }
+       if ts.Unix() < 0 || ts.Unix() >= maxUnixtime {
+               return dt, true
+       }
        return datetime.NewDateTimeFromStd(ts.In(to)), true
 }
```